### PR TITLE
Feature - Update Person Model and AddCommand

### PR DESCRIPTION
**Pull request by cheehongw**
Person model now can contain a list of `SocialMedia` models. 

> A `SocialMedia` is a class that contains a `PlatformName` and `PlatformDescription`, much like a key-value pair. 
>`PlatformName` and `PlatformDescription` are classes with a single `String` field and some not-very-interesting methods (`equals`, `toString`, `getValue`, etc).

In addition, address field in Person is being phased out. This, combined with the addition of `SocialMedia`, necessitated changes in Person's constructor, and associated classes that depend on Person's constructor.

Moreover, storage classes have been updated to store `SocialMedia` model.

**The changes are detailed as follows:**

+ Update `AddCommand` to accept new `Person` constructor
+ Update `SampleDataUtil` to accept new `Person` constructor
+ Update `JsonAdaptedPerson` to accept new `Person` constructor
+ Add `JsonAdaptedSocialMedia` to parse and store `SocialMedia` models
+ Add new getters in `SocialMedia, PlatformName, PlatformDescription`

**Tests**:
- Tests are also updated to match new `Person` constructor, and removal of tests that are related to the address field.
- ~~❗ Could not figure out the one test case where edit command is failing.~~ fixed in [e603c44](https://github.com/AY2122S2-CS2103-W16-2/tp/pull/77/commits/e603c44b832ff4f283dc3bdffa72ff26b786b284)

What's missing:
- Add UI to display `SocialMedia` models. (see TODO comment in `PersonCard.java` @chriswengqi 😃)

### CODE REVIEW

**Review by zihaowrez — COMMENTED** [review]
I think can just delete all the lines about address instead of commenting out.

The rest looks good.

**Comment by cheehongw** [comment]
> I think can just delete all the lines about address instead of commenting out.
> 
> The rest looks good.

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 1]
```diff
@@ -38,6 +38,7 @@
     private Label address;
     @FXML
     private Label email;
+    //TODO Add UI components to handle Social Media
```
Comment (chriswengqi) [inline src/main/java/seedu/address/ui/PersonCard.java:41]: Noted! Will do so!

**Review by chriswengqi — COMMENTED** [review]
LGTM! I will update the UI components of `PersonCard`

**Review by chriswengqi — APPROVED** [review]

**Pull request merged**